extras/videoDrivers/SDL2/VideoSDL2.cpp

```diff
@@ -213,29 +229,22 @@ void VideoSDL2::PrintError(const std::string& msg) const
 
 void VideoSDL2::HandlePaste()
 {
-#ifdef _WIN32
-    if(!IsClipboardFormatAvailable(CF_UNICODETEXT))
+    if(!SDL_HasClipboardText())
         return;
 
-    OpenClipboard(nullptr);
-
-    HANDLE hData = GetClipboardData(CF_UNICODETEXT);
-    const wchar_t* pData = (const wchar_t*)GlobalLock(hData);
+    SDL_memory<char> rawText(SDL_GetClipboardText());
+    if(!rawText)
+        return;
+    if(*rawText == '\0') // empty string indicates error
```

Comment:
I don't see a null-pointer getting returned here per the documentation. So I'd say combine them:
```suggestion
    if(!rawText || *rawText == '\0')
```

Reply:
And constructing a string from a `nullptr` isn't valid anymore. Let's not make the C++ upgrade path more difficult.

Reply:
Being cautious is good. I just wanted to not spend an extra 2 lines and a comment on something that doesn't/shouldn't happen (the SDL doc is pretty clear on that)   
Hence combine the NULL and the `\0` check into a single condition treating both as an error as-if it was documented like that.

Reply:
Well, the SDL doc is clearly lying. It's not magic. They can't return a free-able non-null pointer in an OOM condition. Unless it just aborts in that case, which isn't documented.

```c
char *SDL_GetClipboardText(void)
{
    SDL_VideoDevice *_this = SDL_GetVideoDevice();

    if (_this == NULL) {
        SDL_SetError("Video subsystem must be initialized to get clipboard text");
        return SDL_strdup("");
    }

    if (_this->GetClipboardText) {
        return _this->GetClipboardText(_this);
    } else {
        const char *text = _this->clipboard_text;
        if (text == NULL) {
            text = "";
        }
        return SDL_strdup(text);
    }
}

char *SDL_strdup(const char *string)
{   
    size_t len = SDL_strlen(string) + 1;
    char *newstr = (char *)SDL_malloc(len);
    if (newstr) {
        SDL_memcpy(newstr, string, len);
    }       
    return newstr;
}
```

I rest my case.

Reply:
> Well, the SDL doc is clearly lying.

I thought so, too. Good job finding the relevant code :+1:

My point is that treating both `!text` and `*text == '\0' the same as "an error occurred" makes more sense than using 2 separated conditionals requiring 2 comments explaining each of them, i.e.:

```
    if(some condition or multiple)
        PrintError(SDL_GetError());
```

doesn't require any further elaboration that this is a check for an error and avoids tinkering about why one part would just return while the other prints etc. In one word: Simplicity :)

Reply:
But calling `SDL_GetError()` simply isn't valid if `!text`. That is my concern.
```cpp
 if(!text || *text == '\0')
        PrintError(text ? SDL_GetError() : "Paste failed.");
```

Is that an acceptable compromise?